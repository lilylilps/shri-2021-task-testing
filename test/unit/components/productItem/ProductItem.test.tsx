import React from "react";
import renderer from 'react-test-renderer';
import { Router } from "react-router";
import { Provider } from "react-redux";
import { createMemoryHistory } from "history";

import { it, describe, expect } from "@jest/globals";

import { initStore } from "../../../../src/client/store";
import { ProductItem } from "../../../../src/client/components/ProductItem";
import { ExampleApi, CartApi } from "../../../../src/client/api";


import { Store } from "redux";

describe('проверка компонента productItem', () => {

    let api: ExampleApi;
    let cart: CartApi;
    let store: Store;

    beforeAll(() => {
        api = new ExampleApi('');
        cart = new CartApi();
        store = initStore(api, cart);
    });

    const product = {
        id: 1,
        name: "shorts",
        price: 200,
    }

    it('snapshot', () => {
        const history = createMemoryHistory({
            initialEntries: ['/catalog'],
            initialIndex: 0
        });

        const tree = renderer.create(
            <Router history={history}>
                <Provider store={store}>
                    <ProductItem product={product} />
                </Provider>
            </Router>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});