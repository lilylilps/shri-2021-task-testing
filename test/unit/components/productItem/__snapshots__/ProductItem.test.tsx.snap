// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка компонента productItem snapshot 1`] = `
<div
  className="ProductItem card w-100 mb-4"
  data-testid={1}
>
  <img
    className="Image card-img-top"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
  />
  <div
    className="card-body"
  >
    <h5
      className="ProductItem-Name card-title"
    >
      shorts
    </h5>
    <p
      className="ProductItem-Price card-text"
    >
      $
      200
    </p>
    <a
      className="ProductItem-DetailsLink card-link"
      href="/catalog/1"
      onClick={[Function]}
    >
      Details
    </a>
  </div>
</div>
`;
